src/state/image-dataset/json-dataset/index.ts

```diff
@@ -92,31 +89,111 @@ class JsonRequest implements ImageDataset {
 
         // for now this particular file must be kept up to date.
         // dev-aics-dtp-001/cfedata is the designated repository of internal cfe data sets
+        // this datasets.json contains a list of the names of subdirectories (strings),
+        // each of which is expected to contain a dataset.json file
         this.listOfDatasetsDoc =
-            "http://dev-aics-dtp-001.corp.alleninstitute.org/cfedata/datasets.json";
+            "http://dev-aics-dtp-001.corp.alleninstitute.org/cfedata/cell-feature-data/datasets.json";
     }
 
     public getAvailableDatasets = () => {
-        return axios
-            .get(`${this.listOfDatasetsDoc}`)
-            .then((metadata: AxiosResponse) => metadata.data);
+        //const megasets: Megaset[] = [];
+        return axios.get(`${this.listOfDatasetsDoc}`).then((metadata: AxiosResponse) => {
+            const datadir = this.listOfDatasetsDoc.substring(
+                0,
+                this.listOfDatasetsDoc.lastIndexOf("/")
+            );
+
+            return Promise.all(
+                (metadata.data as string[]).map((datasetdir) => {
```

Comment:
Both the top-level function and the argument to `map` could become `async`. By no means necessary, but would save a couple indents.

Reply:
I'm going to hold off on this change as the diff is already a bit much.

Reply:
agreed, I went ahead and fixed that